src/rust/engine/src/intrinsics.rs

```diff
@@ -645,3 +653,46 @@ fn interactive_process(
     Ok(result)
   }.boxed()
 }
+
+use process_execution::docker::DockerOnceCell;
+fn docker_resolve_image(
+  context: Context,
+  args: Vec<Value>,
+) -> BoxFuture<'static, NodeResult<Value>> {
+  async move {
+    let types = &context.core.types;
+    let docker_resolve_image_result = types.docker_resolve_image_result;
+
+    let image_name = Python::with_gil(|py| {
+      let py_docker_request = (*args[0]).as_ref(py);
+      let image_name: String = externs::getattr(py_docker_request, "image_name").unwrap();
+      image_name
+    });
+
+    // TODO: Store this somewhere as a singleton?
+    let docker_cell = DockerOnceCell::new();
```

Comment:
Maybe this `DockerOnceCell` should be shared with `docker::CommandRunner` as a global?

Reply:
That or the entire image pull cache, yea?

Reply:
Probably yeah. I'll probably want to include some sort of "image pull scope" key to manage applying image pull policy correctly for sessions once that image pull cache is global. But to start, it doesn't need such support.